modules/l4proxy/upstream.go

```diff
@@ -46,6 +57,12 @@ type Upstream struct {
 	// have before being marked as unhealthy (if > 0).
 	MaxConnections int `json:"max_connections,omitempty"`
 
+	// Is dial a dynamic upstream
+	// If true, skip address parsing in Upstream.provision
+	// then parse the address after replacing all placeholders in Handler.dialPeers.
+	// (support dynamic port)
+	Dynamic bool `json:"dynamic,omitempty"`
```

Comment:
No need for a new field. I think it's better that the need for delayed parsing is inferred if the dial address contains `{` and `}`. Pretty sure that's what we did in Caddy `reverse_proxy` handler.

Reply:
The condition has been changed to the address contains `{` and `}` and the new field has been removed. Thank you.